src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -74,16 +74,24 @@ private long computeCountUsingStaticScale()
         return getRowCountForScale(1);
     }
 
-    private long computeCountUsingLogScale(int scale)
+    private long computeCountUsingLogScale(double scale)
     {
         int scaleSlot = getScaleSlot(scale);
         long delta = getRowCountForScale(DEFINED_SCALES[scaleSlot]) - getRowCountForScale(DEFINED_SCALES[scaleSlot - 1]);
-        float floatOffset = (float) (scale - DEFINED_SCALES[scaleSlot - 1]) / (float) (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
+        double floatOffset = (scale - DEFINED_SCALES[scaleSlot - 1]) / (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
 
-        return (int) (floatOffset * (float) delta) + getRowCountForScale(DEFINED_SCALES[0]);
+        long baseRowCount = 0;
+        if (scale < 1.0) {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[0]);
+        }
+        else {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[1]);
+        }
+        long count = (long) (floatOffset * (float) delta) + baseRowCount;
```

Comment:
I don't understand. The count has to be converted to a long because we use longs to store the number of rows. The conversion to long happens as late as possible. Am I misunderstanding? I guess previously we would cast it to an int here but then the cast to long would have happened implicitly since the return type of the method has always been long.